src/Elmish/HTML/Events/Internal.purs

```diff
@@ -0,0 +1,66 @@
+-- | Internal implementation details. See comments on `Elmish.HTML.Events` instead.
+module Elmish.HTML.Events.Internal
+  ( REventBase
+  , RKeyboardEvent
+  , RModifierKeys
+  , RMouseEvent
+  , RSyntheticEvent
+  , RTouchEvent
+  )
+  where
+
+import Web.DOM (Element)
+import Web.Event.Internal.Types (Event)
+import Type.Row (type (+))
+
+type RSyntheticEvent = { | REventBase () }
+
+type RKeyboardEvent =
+  { charCode :: Int
+  , key :: String
+  , keyCode :: Int
+  , locale :: String
+  , location :: Int
+  , repeat :: Boolean
+  , which :: Int
+  | RModifierKeys + REventBase + ()
```

Comment:
What does `+ ()` do at the type level? Or: Why is it needed?

Reply:
This is type-level stuff.

The definition of `+` is that `f + a = f a`, where `f :: Row -> Row` (a partial row) and `a :: Row` (a closed row), and the idea is to combine partial rows without nesting. This does not make sense at all with just a single application - why would I write `f + a` when `f a` is much shorter? But it becomes much nicer when there are more than one partial row involved:

```
f + g + a = f (g a)
```

So, for example:

```
type F a = ( x :: Int | a )
type G a = ( y :: Int | a )

type Combined = F + G + ( z :: Int )
-- Combined = ( x :: Int, y :: Int, z :: Int )
```

So, to a human eye, the plus looks like "combining" the rows together. Adding them, in a sense.

The caveat, however, is that the very last addend must be a closed row. You can't write just `F + G`, because the result would have to be something like `\a -> F (G a)`, but you can't have lambda-expressions at type level.

On this particular line, I need to combine `RModifierKeys` with `REventBase`, but I don't have anything else to add, so I use an empty row at the very end.

Reply:
Awesome!Thanks for the in-depth explanation 🙇🏼‍♂️